lib/config/config-file.js

```diff
@@ -210,6 +244,8 @@ function loadConfigFile(filePath) {
             default:
                 config = loadLegacyConfigFile(filePath);
         }
+    } else if (filePath.indexOf("eslint-plugin") >= 0) {
```

Comment:
I'm still concerned about this check, as it would also pass for "whatever-eslint-plugin". I think it's safer to reattach " plugin:" before calling this function and checking that it's index is 0.